### PR TITLE
Added option to use seconds instead of number of lines while reading dmesg output

### PR DESCRIPTION
**Pull request by epierotto**
## Pull Request Checklist

**Is this in reference to an existing issue?**

https://github.com/sensu-plugins/sensu-plugins-hardware/issues/8

#### General

- [x] Update Changelog following the conventions laid out on [Our CHANGELOG Guidelines](https://github.com/sensu-plugins/community/blob/master/HOW_WE_CHANGELOG.md)

- [x] Update README with any necessary configuration snippets

- [ ] Binstubs are created if needed

- [x] RuboCop passes

- [ ] Existing tests pass

#### New Plugins

- [ ] Tests

- [ ] Add the plugin to the README

- [ ] Does it have a complete header as outlined [here](http://sensu-plugins.io/docs/developer_guidelines.html#coding-style)

#### Known Compatablity Issues

### CODE REVIEW

**Comment by majormoses** [comment]
Thanks for your contribution to Sensu plugins! Without people like you submitting PRs we couldn't run the project. I will review it shortly.

File: bin/check-hardware-fail.rb [hunk 1]
```diff
@@ -43,21 +43,102 @@ class CheckHardwareFail < Sensu::Plugin::Check::CLI
...
+    lines = `#{cmd}`.lines
     unknown 'Command execution failed!' unless $CHILD_STATUS.success?
+
+    if config[:seconds]
+      uptime = File.read('/proc/uptime').split(' ').first.to_i
+      seconds_limit = uptime - config[:seconds]
+    end
+
+    # Option --invert
+    lines.reverse! unless config[:invert]
```
[the remark this answers is not in the document]
Comment (majormoses) [reply bin/check-hardware-fail.rb]: ah that makes sense can you please leave a comment above so this is clear later? I am not sure I really see the invert option making a ton of sense either except if maybe you are looking for only issues while attempting to initialize new hardware as it is boot up and do not care about the degradation? It does indeed seem odd.

File: bin/check-hardware-fail.rb [hunk 2]
```diff
@@ -43,21 +43,102 @@ class CheckHardwareFail < Sensu::Plugin::Check::CLI
...
+
+    output = []
+
+    lines.each_with_index do |line, index|
+      break if config[:lines] && index >= config[:lines]
```
[the remark this answers is not in the document]
Comment (majormoses) [reply bin/check-hardware-fail.rb]: good catch.

File: bin/check-hardware-fail.rb [hunk 3]
```diff
@@ -43,21 +43,104 @@ class CheckHardwareFail < Sensu::Plugin::Check::CLI
...
     unknown 'Command execution failed!' unless $CHILD_STATUS.success?
+
+    if config[:seconds]
+      uptime = File.read('/proc/uptime').split(' ').first.to_i
+      seconds_limit = uptime - config[:seconds]
+    end
+
+    # Option --invert
+    # Flipping the array to start the iteration from the last line (newest entry)
+    # unless `--invert` which means that you want to start reading from the first line (oldest entry)
```
Comment (epierotto) [inline bin/check-hardware-fail.rb]: @majormoses I've added this comment here to explain why we flip the array

File: bin/check-hardware-fail.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (epierotto) [reply bin/check-hardware-fail.rb]: I've tested the condition today and is already working as expected with `--invert`, nothing to change there.

**Comment by epierotto** [comment]
Adding artefacts:

# Linux:
```
check-hardware-fail.rb --lines 100 --facility daemon,auth --level err,warn,notice --kernel -q 'BOOT_IMAGE'
CheckHardwareFail OK: OK
```

```
check-hardware-fail.rb --seconds 100  --facility daemon,auth --level err,warn,notice --kernel  -q 'BOOT_IMAGE'
CheckHardwareFail OK: OK
```

```
check-hardware-fail.rb --lines 100 --facility daemon,auth --level err,warn,notice --kernel --invert -q 'BOOT_IMAGE'
CheckHardwareFail CRITICAL: Problem Detected: [    0.000000] Kernel command line: BOOT_IMAGE=/boot/vmlinuz-3.10.0-693.5.2.el7.x86_64 root=UUID=29342a0b-e20f-4676-9ecf-dfdf02ef6683 ro console=tty0 crashkernel=auto console=ttyS0,115200 net.ifnames=0
```

### Incompatible options
```
check-hardware-fail.rb --seconds 100 --facility daemon,auth --level err,warn,notice --kernel --invert -q 'BOOT_IMAGE'
CheckHardwareFail UNKNOWN: The use of the options --seconds and [--invert] is incompatible.
```

```
check-hardware-fail.rb --seconds 100 --lines 100 --facility daemon,auth --level err,warn,notice --kernel -q 'BOOT_IMAGE'
CheckHardwareFail UNKNOWN: The use of the options --seconds and [--lines] is incompatible.
```

```
check-hardware-fail.rb --seconds 100 --lines 100 --facility daemon,auth --level err,warn,notice --kernel --invert -q 'BOOT_IMAGE'
CheckHardwareFail UNKNOWN: The use of the options --seconds and [--lines --invert] is incompatible.
```

### Incompatible dmesg facility or level
```
check-hardware-fail.rb --seconds 100  --facility daemon,auth --level err,warn,notice,bananas --kernel  -q 'BOOT_IMAGE'
CheckHardwareFail UNKNOWN: Invalid dmesg level requested: ["bananas"]
```

```
check-hardware-fail.rb --seconds 100  --facility daemon,auth,apple --level err,warn,notice --kernel  -q 'BOOT_IMAGE'
CheckHardwareFail UNKNOWN: Invalid dmesg facility requested: ["apple"]
```

# OSX:
```
check-hardware-fail.rb --lines 100 -q 'BOOT_IMAGE'
CheckHardwareFail OK: OK
```

```
check-hardware-fail.rb --lines 100 --invert -q 'mDestroyDeviceCalled'
CheckHardwareFail CRITICAL: Problem Detected: queuedAckedPackets = 0, mDestroyDeviceCalled = FALSE -- hostController = 0x6000 ****
```

```
check-hardware-fail.rb --lines 100 -q 'mDestroyDeviceCalled'
CheckHardwareFail CRITICAL: Problem Detected: **** [IOBluetoothDevice][decrementNumberOfOutstandingPacketsBy] -- Handle 0x000c -- remove ACL Timer 0xffffff8054155c00 for ACL packet 0xffffff8065157d00 -- numOutstandingHWPackets = 1, mNumNotDequeuedAckedPackets = 0, mDestroyDeviceCalled = FALSE -- hostController = 0x6000 ****
```

### Incompatible options
```
check-hardware-fail.rb --seconds 100 -q 'mDestroyDeviceCalled'
CheckHardwareFail UNKNOWN: Incompatible options: [--seconds] for OS darwin14.5.0
```

```
check-hardware-fail.rb --seconds 100  --lines 100 -q 'mDestroyDeviceCalled'
CheckHardwareFail UNKNOWN: Incompatible options: [--seconds] for OS darwin14.5.0
```

```
check-hardware-fail.rb --seconds 100  --facility daemon,auth --level err,warn,notice --kernel  -q 'mDestroyDeviceCalled'
CheckHardwareFail UNKNOWN: Incompatible options: [--seconds --facility --level --kernel] for OS darwin14.5.0
```

**Comment by majormoses** [comment]
released: https://rubygems.org/gems/sensu-plugins-hardware/versions/1.3.0

**Pull request merged**